### DISCUSSION TITLE
Privacy / confidentiality concern

### DISCUSSION THREAD

**Discussion by vmeschanov** [keplergl/kepler.gl#2627] [Q&A] [answered]
Hello,
I wonder whether I can use Kepler.gl with my data totally sitting on my servers and not leaving it? I don't want to upload any datasets to Mapbox or anywhere else.
I see, that I can source data from my server in the JavaScript code, but does Kepler.gl or Mapbox send any part of it to the outside? I don't need geo-coding, I have exact coordinates.
Thank you !

**Comment by ibgreen** [comment, marked as answer]
Yes, kepler only loads your data into the user's browser, it does not send it anywhere.

**Reply by vmeschanov** [reply]
Thank you!
